sdk/js/src/service/applications.js

```diff
@@ -52,93 +55,82 @@ class Applications {
     this.updateById = this.updateById.bind(this)
     this.create = this.create.bind(this)
     this.deleteById = this.deleteById.bind(this)
-    this.withId = this.withId.bind(this)
     this.getRightsById = this.getRightsById.bind(this)
   }
 
+  _responseTransform (response, single = true) {
+    return Marshaler[single ? 'unwrapApplication' : 'unwrapApplications'](
+      response,
+      this._proxy
+        ? app => new Application(this, app, false)
+        : undefined
+    )
+  }
+
   // Retrieval
 
   async getAll (params) {
-    const result = await this._api.ApplicationRegistry.List({ query: params })
+    const response = await this._api.ApplicationRegistry.List({ queryParams: params })
 
-    return Marshaler.unwrapApplications(
-      result,
-      this._applicationTransform
-    )
+    return this._responseTransform(response, false)
   }
 
   async getById (id) {
-    const result = await this._api.ApplicationRegistry.Get({
-      route: { 'application_ids.application_id': id },
+    const response = await this._api.ApplicationRegistry.Get({
+      routeParams: { 'application_ids.application_id': id },
     })
 
-    return Marshaler.unwrapApplication(
-      result,
-      this._applicationTransform
-    )
+    return this._responseTransform(response)
   }
 
   async getByOrganization (organizationId) {
-    const result = this._api.ApplicationRegistry.List({
-      route: { 'collaborator.organization_ids.organization_id': organizationId },
+    const response = this._api.ApplicationRegistry.List({
+      routeParams: { 'collaborator.organization_ids.organization_id': organizationId },
     })
 
-    return Marshaler.unwrapApplications(
-      result,
-      this._applicationTransform
-    )
+    return this._responseTransform(response)
   }
 
   async getByCollaborator (userId) {
-    const result = this._api.ApplicationRegistry.List({
-      route: { 'collaborator.user_ids.user_id': userId },
+    const response = this._api.ApplicationRegistry.List({
+      routeParams: { 'collaborator.user_ids.user_id': userId },
     })
 
-    return Marshaler.unwrapApplications(
-      result,
-      this._applicationTransform
-    )
+    return this._responseTransform(response)
   }
 
   async search (params) {
-    const result = await this._api.EntityRegistrySearch.SearchApplications({
-      query: params,
+    const response = await this._api.EntityRegistrySearch.SearchApplications({
+      queryParams: params,
     })
-    return Marshaler.unwrapApplications(
-      result,
-      this._applicationTransform
-    )
+
+    return this._responseTransform(response, false)
   }
 
   // Update
 
-  async updateById (id, patch, mask = Marshaler.fieldMaskFromPatch(patch)) {
-    const result = await this._api.ApplicationRegistry.Update({
-      route: {
-        'application.ids.application_id': id,
-      },
+  async updateById (id, patch, selector) {
+    const fieldMask = selector
```

Comment:
why? do you change anything in the Application service? this doesnt seem to be related to devices

Reply:
Agreed. This is not necessary and I will remove it. However, we actually need to change the `getById` method to accept selectors (as intermediate to field masks), otherwise, we won't get e.g. `name` or `description` field by default, which is the case at the moment. I would add that in this PR as well, as it is a small change.